### DISCUSSION TITLE
Is data leak issue relevant on classic NextJs Page-Router, not App Router?

### DISCUSSION THREAD

**Discussion by chiralium** [pmndrs/zustand#2258] [Q&A]
#2200 in this discussion I've reliaze that Zustand usage with NextJs require the specific mechanism. Anyone can explain me is I'm in the risk when use Zustand on Page-Router NextJs application? The linked issue has describe about only page router.

**Comment by dbritto-dev** [comment]
The same for App Router

**Comment by chiralium** [comment]
> The same for App Router

I don’t understand, is this bug relevant for page router?

**Reply by dbritto-dev** [reply]
Actually, isn't a bug is an expected behavior even if you use client components (`'use client'`) the components are processed on the server so you should always relay on `useEffect()`. The issue with server and client components is there're just a way for code/bundle splitting. Both are processed on the server but just client components run on the client side

**Reply by chiralium** [reply]
I'm using NextJs without AppRouter and RSC. Tell me, will I have problems using Zustand

**Comment by dai-shi** [comment]
I wonder if it helps, but it's basically same as a global variable.
```js
let globalCounter = 0;
```

**Reply by chiralium** [reply]
Is there any solution how to deal with this? I like no-store actions, but now, I need use react-context to share stores between components.

**Reply by dai-shi** [reply]
Well, global variable is global, so no solutions to it.
[Jotai](https://github.com/pmndrs/jotai) might feel better in this case.
